improve documentation for tidy methods (#1230)
tidymodels/recipes: NEWS.md

* Document that `update_role()`, `add_role()` and `remove_role()` are applied before steps and checks. (#778)

> * Documentation for tidy methods for all steps have been added when missing and improved to describe whole outcome. (#936)

Reviewer: ```suggestion
* Documentation for tidy methods for all steps has been added when missing and improved to describe the return value more accurately. (#936)
```
